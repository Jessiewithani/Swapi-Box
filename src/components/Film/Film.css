.filmContainer {
  background-color: black;
  width: 100%;
  height: 100vh;
  overflow:hidden;
}

.filmContainer {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 600px; /* alllows us to transfrom the text 3D, from how wide it comes from the z-angle */
  color: white;
  font-size: 300%;
  font-family: 'Dunkin';
  letter-spacing: 6px;
  line-height: 140%;
  text-align: justify;
}

.crawl {
  position: relative;
  top: -100px; /* makes the text fully off the screen */
  transform-origin: 50% 100%; /* allows us to change the position of transformed elements */
  animation: crawl 30s linear; /* plays the crawl for 1 minute */
}


/* we will have to focus on the z index which comes at an angle */

@keyframes crawl {
   0% { /* this is where the text is closest to the viewer */
     top: 0; /* this is what makes the text start below the screen */
     transform: rotateX(40deg) translateZ(0); /* rotating the text 20 degrees but still keeping it closer to the viewer */
  }
   100% { /* this is where the tet is far away from the viewer and goes off the screen at the top*/
     top: -4000px; /* makes sure the text is completely off the screen */
     transform: rotateX(25deg) translateZ(-2500px); /* moves the text further and further away from the viewer */
  }
}
