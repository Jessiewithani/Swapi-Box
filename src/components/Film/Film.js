import React, { Component } from 'react';
import './Film.css'

class Film extends Component {
    constructor() {
        super()
        this.state = {
            film: []
        }
    }
    componentDidMount = () => {
        let randomNumber = Math.floor(Math.random() * 7)
        fetch('https://swapi.co/api/films')
            .then(response => response.json())
            .then(data => {
                return data;
            })
            .then(data => this.setState({ film: data.results[randomNumber] }))

    }
    render() {
        return (
            <aside>
                <h2>{this.state.film.title}</h2>
                <h2>{this.state.film.release_date}</h2>
                <h2>{this.state.film.opening_crawl}</h2>
            </aside>
        )
    }
}

export default Film;