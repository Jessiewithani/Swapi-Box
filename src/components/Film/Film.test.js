import React from 'react';
import { shallow } from 'enzyme';
import Film from './Film';

describe('Film', () => {
  it('should match the snapshot', () => {
    const wrapper= shallow(<Film/>)
    expect(wrapper).toMatchSnapshot()
  });
});