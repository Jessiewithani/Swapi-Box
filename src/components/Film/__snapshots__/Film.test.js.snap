// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film should match the snapshot 1`] = `
<aside>
  <h2 />
  <h2 />
  <h2 />
</aside>
`;
